internal/pkg/service/stream/storage/level/local/encoding/encoder/config.go

```diff
@@ -1,12 +1,20 @@
-package format
+package encoder
 
 // Config configures the local writer.
 type Config struct {
 	Concurrency int `configKey:"concurrency" configUsage:"Concurrency of the format writer for the specified file type. 0 = auto = num of CPU cores" validate:"min=0,max=256"`
+	// Factory creates a high-level writer for the storage.FileType, for example storage.FileTypeCSV.
+	// A custom implementation can be useful for tests.
+	Factory Factory
 }
 
-// ConfigPatch is same as the Config, but with optional/nullable fields.
-// It may be part of a Sink definition to allow modification of the default configuration.
 type ConfigPatch struct {
 	Concurrency *int `json:"concurrency,omitempty"`
 }
+
+func NewConfig() Config {
+	return Config{
+		Concurrency: 0, // 0 = auto = CPU cores
+		Factory:     DefaultFactory{},
```

Comment:
There is issue with the `assert.Equal`, with comparing references to functions, so I used interfaces and structs.

Read more; https://github.com/stretchr/testify/issues/182